src/main/java/com/arcanewarrior/disguises/events/DisguiseEvents.java

```diff
@@ -81,14 +85,80 @@ private void handlePlayerPacketInput(PlayerPacketEvent event) {
             }
         }
     }
+
+    private void playerTeleport(PlayerPacketOutEvent event) {
+        if (event.getPacket() instanceof EntityTeleportPacket packet) {
+            Disguise disguise = parentManager.getPlayerDisguise(event.getPlayer());
+            if (disguise != null) {
+                EntityTeleportPacket newPacket = new EntityTeleportPacket(disguise.getEntityId(), packet.position(), packet.onGround());
+                PacketUtils.sendGroupedPacket(disguise.getViewers(), newPacket);
+            }
+        }
+    }
+
+    private void playerSpawn(PlayerPacketOutEvent event) {
+        if (event.getPacket() instanceof SpawnPlayerPacket packet) {
+            Player player = MinecraftServer.getConnectionManager().getPlayer(packet.playerUuid());
+            if (player != null) {
+                Disguise disguise = parentManager.getPlayerDisguise(player);
+                if (disguise != null) {
+                    player.removeViewer(event.getPlayer());
+                    disguise.addViewer(event.getPlayer());
+                }
+            }
+        }
+    }
+
+    private void playerEffect(PlayerPacketOutEvent event) {
+        if (event.getPacket() instanceof EntityEffectPacket packet) {
+            Player player = event.getPlayer();
+            Disguise disguise = parentManager.getPlayerDisguise(player);
+            if (disguise != null) {
+                EntityEffectPacket newPacket = new EntityEffectPacket(disguise.getEntityId(), packet.potion(), packet.factorCodec());
+                PacketUtils.sendGroupedPacket(disguise.getViewers(), newPacket);
+            }
+        }
+        if (event.getPacket() instanceof RemoveEntityEffectPacket packet) {
+            Player player = event.getPlayer();
+            Disguise disguise = parentManager.getPlayerDisguise(player);
+            if (disguise != null) {
+                RemoveEntityEffectPacket newPacket = new RemoveEntityEffectPacket(disguise.getEntityId(), packet.potionEffect());
+                PacketUtils.sendGroupedPacket(disguise.getViewers(), newPacket);
+            }
+        }
+    }
+
+    private void playerVelocity(PlayerPacketOutEvent event) {
+        if (event.getPacket() instanceof EntityVelocityPacket packet) {
+            Player player = event.getPlayer();
+            Disguise disguise = parentManager.getPlayerDisguise(player);
+            if (disguise != null) {
+                EntityVelocityPacket newPacket = new EntityVelocityPacket(packet.entityId(), packet.velocityX(), packet.velocityY(), packet.velocityZ());
+                PacketUtils.sendGroupedPacket(disguise.getViewers(), newPacket);
+            }
+        }
+    }
+
+    private void playerPassengers(PlayerPacketOutEvent event) {
+        if (event.getPacket() instanceof SetPassengersPacket packet) {
+            Player player = event.getPlayer();
+            if (packet.vehicleEntityId() == packet.getId()) {
```

Comment:
I can see what this is going for (add passengers to disguise, but this comparision doesn't work (it compares entity id to the packet's internal id).